### DISCUSSION TITLE
The onBlur and onChange functions comming from register function are the exact same

### DISCUSSION THREAD

**Discussion by omardevelopsGit** [react-hook-form/react-hook-form#12196] [Q&A]
When logging register(...args) to the console the onBlur and onChange are the exact same.
Codesandbox: https://codesandbox.io/p/sandbox/admiring-sky-sqhrp3?file=%2Fsrc%2FInput.jsx%3A72%2C1
![image](https://github.com/user-attachments/assets/975bed03-a3ad-4ace-8dc0-ec91518a7389)

**Comment by BrendanC23** [comment]
Based on the code, this looks intentional. Is it causing you an actual problem? Here's the relevant code:

https://github.com/react-hook-form/react-hook-form/blob/81f472ced10e03e11932b35bec1b75e02dba5718/src/logic/createFormControl.ts#L1066

https://github.com/react-hook-form/react-hook-form/blob/81f472ced10e03e11932b35bec1b75e02dba5718/src/logic/createFormControl.ts#L693-L826

Notably,

```ts
const isBlurEvent =  event.type === EVENTS.BLUR || event.type === EVENTS.FOCUS_OUT;
```

The same function is called for both `change` and `blur`, but the function  checks the event type and acts differently in each case.
